### PR TITLE
feat: add requireAuthentication directive

### PR DESCRIPTION
**Pull request by Aenimus**
## Motivation and Context

Allow the user to configure an operation to require authentication through a directive.

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs-website/src/pages/docs/directives-reference/require-authentication-directive.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: '@requireAuthentication Directive'
+description: Configure an operation to require authentication
+---
+
+{% callout type="warning" %}
+This directive takes priority over all other operation authentication configs.
```
Comment (fiam) [inline docs-website/src/pages/docs/directives-reference/require-authentication-directive.md:7]: `configs` -> `configurations`?

**Review by fiam — APPROVED** [review]
LGTM

File: docs-website/src/pages/docs/directives-reference/require-authentication-directive.md (hunk 1 above)
Comment (Aenimus) [reply docs-website/src/pages/docs/directives-reference/require-authentication-directive.md]: Yeah, I thought about this myself. Changed. Thanks.

Comment (fiam) [reply docs-website/src/pages/docs/directives-reference/require-authentication-directive.md]: Thanks!

**Review by jensneuse — APPROVED** [review]

**Pull request merged**